modules/repofiles/delete_test.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package repofiles
+
+import (
+	"testing"
+
+	"code.gitea.io/gitea/models"
+	"code.gitea.io/gitea/modules/test"
+	api "code.gitea.io/sdk/gitea"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func getDeleteRepoFileOptions(repo *models.Repository) *DeleteRepoFileOptions {
+	return &DeleteRepoFileOptions{
+		LastCommitID: "",
+		OldBranch:    repo.DefaultBranch,
+		NewBranch:    repo.DefaultBranch,
+		TreePath:     "README.md",
+		Message:      "Deletes README.md",
+		SHA:          "4b4851ad51df6a7d9f25c979345979eaeb5b349f",
+		Author:       nil,
+		Committer:    nil,
+	}
+}
+
+func getExpectedDeleteFileResponse() *api.FileResponse {
+	return &api.FileResponse{
+		Content: nil,
+		Commit: &api.FileCommitResponse{
+			CommitMeta: api.CommitMeta{
+				URL: "https://try.gitea.io/api/v1/repos/user2/repo1/git/commits/65f1bf27bc3bf70f64657658635e66094edbcb4d",
+				SHA: "65f1bf27bc3bf70f64657658635e66094edbcb4d",
+			},
+			HTMLURL: "https://try.gitea.io/user2/repo1/commit/65f1bf27bc3bf70f64657658635e66094edbcb4d",
+			Author: &api.CommitUser{
+				Identity: api.Identity{
+					Name:  "user1",
+					Email: "address1@example.com",
+				},
+				Date: "2017-03-19T20:47:59Z",
+			},
+			Committer: &api.CommitUser{
+				Identity: api.Identity{
+					Name:  "Ethan Koenig",
+					Email: "ethantkoenig@gmail.com",
+				},
+				Date: "2017-03-19T20:47:59Z",
+			},
+			Parents: []*api.CommitMeta{},
+			Message: "Initial commit\n",
+			Tree: &api.CommitMeta{
+				URL: "https://try.gitea.io/api/v1/repos/user2/repo1/git/trees/2a2f1d4670728a2e10049e345bd7a276468beab6",
+				SHA: "2a2f1d4670728a2e10049e345bd7a276468beab6",
+			},
+		},
+		Verification: &api.PayloadCommitVerification{
+			Verified:  false,
+			Reason:    "",
+			Signature: "",
+			Payload:   "",
+		},
+	}
+}
+
+func TestDeleteRepoFile(t *testing.T) {
+	// setup
+	models.PrepareTestEnv(t)
+	ctx := test.MockContext(t, "user2/repo1")
+	ctx.SetParams(":id", "1")
+	test.LoadRepo(t, ctx, 1)
+	test.LoadRepoCommit(t, ctx)
+	test.LoadUser(t, ctx, 2)
+	test.LoadGitRepo(t, ctx)
+	repo := ctx.Repo.Repository
+	doer := ctx.User
+	opts := getDeleteRepoFileOptions(repo)
+
+	// test - Delete the README.md file
+	// actual test
+	fileResponse, err := DeleteRepoFile(repo, doer, opts)
```

Comment:
If you genuinely have subtests you can label them as such using `t.Run(...)` and run `PrintCurrentTest(t)` we'll get a pretty printed in our integration tests. (As these seem to be very small subtests it's probably better to not bother printing as it will spam our integration test logs.) 

```suggestion
	t.Run("DeleteREADME.md", func (t *testing.T) {
		fileResponse, err := DeleteRepoFile(repo, doer, opts)
		
		// asserts
		assert.Nil(t, err)
		expectedFileResponse := getExpectedDeleteFileResponse()
		assert.EqualValues(t, expectedFileResponse, fileResponse)
	})
```

Not a hard comment just a suggestion.

Reply:
ok, added these subtests.